opencog/query/PatternSCM.cc

```diff
@@ -69,11 +69,17 @@ void PatternSCM::init(void)
 	// BindLink containing variables, a pattern and a rewrite rules.
 	_binders.push_back(new FunctionWrap(bindlink, "cog-bind", "query"));
 
-	// Identical to do_bindlink above, except that it only returns the
+	// Identical to bindlink above, except that it only returns the
 	// first match.
 	_binders.push_back(new FunctionWrap(single_bindlink,
 	                   "cog-bind-single", "query"));
 
+	// Identical to bindlink above, except that it only returns the
+	// first N matches, assuming that N is the first argument and
+	// the second is a BindLink handle.
+	_binders.push_back(new FunctionWrap(first_n_bindlink,
+	                   "cog-bind-first-n", "query"));
+
```

Comment:
Well, and it probably makes sense to also remove the code for `cog-bind` as well, and replace it by
```
(define-public (cog-bind atom) (cog-bind-first-n -1 atom))
```

Notice that I used `-1` in the above instead of unsigned-max.  The reason for this is that I fear bugs due to some crazy mis-match between the C++ UNSIGNED_MAX, and the scheme version of the same. But maybe this is an old-fashioned way of thinking; I should just trust the modern-day systems to do the right thing...

Figuring out unsigned-max in guile might be hard -- its normally arbitrary precision; it has no max....

Your pick in how to handle this. My goal is to minimize the total amount of code; the fewer lines of code the better -- the less confusing things will be.